packages/client/src/create-deployment.ts

```diff
@@ -82,7 +86,34 @@ export default function buildCreateDeployment() {
       };
     }
 
-    const files = await hashes(fileList);
+    // Populate Files -> FileFsRef mapping
+    const workPath = typeof path === 'string' ? path : path[0];
+
+    let files;
+
+    if (clientOptions.archive === ArchiveFormat.Tgz) {
+      debug('Packing tarball');
+      fileList = fileList.map(file => file.replace(workPath, ''));
```

Comment:
Can you inline this on line 99 or do we expect to mutate `fileList` to use it later on?

Reply:
Also, replace might cause problems if it replaces more than one pattern.

Lets use relative path since I think that is the intention:

```suggestion
      const entries = fileList.map(file => path.relative(workPath, file));
```

Reply:
I don't see why the logic can't be inline on line 99